library/firewall_lib.py

```diff
@@ -179,6 +179,40 @@
     HAS_FIREWALLD = False
 
 
+def handle_interface_permanent(
+    zone, item, fw_zone, fw_settings, fw, fw_offline, module
+):
+    if fw_offline:
+        iface_zone_objs = []
+        for zone in fw.config.get_zones():
+            old_zone_obj = fw.config.get_zone(zone)
+            if item in old_zone_obj.interfaces:
+                old_zone_settings = FirewallClientZoneSettings(
+                    fw.config.get_zone_config(old_zone_obj)
+                )
+                old_zone_settings.removeInterface(item)
```

Comment:
You don't need line 201 anymore since you are doing the `removeInterface` on line 193 and that should catch all zones.
I think line 202 should be moved here:
```
             fw.config.set_zone_config(old_zone_obj, old_zone_settings.settings)
```

Reply:
If having both `removeInterface` is all that is needed, are we all set to merge this change in?